syntax = "proto3";

package sqlite_sync;

// The synchroniser service definition.
service Synchroniser {
  // Registers a db
  rpc RegisterDb (RegisterDbRequest) returns (RegisterDbReply) {}

  // Synchronise db statement
  rpc SyncDbStatement (SyncDbStatementRequest) returns (SyncDbStatementReply) {}
}

// The request message containing the db name and other params.
message RegisterDbRequest {
  string name = 1;
}

// The response message containing the registration status
message RegisterDbReply {
  uint32 status = 1;
}

message SyncDbStatementRequest {
  string name = 1;
  uint32 default_db = 2;
  string statement = 3;
}

message SyncDbStatementReply {
  uint32 status = 1;
}